src/common/crypto/platform/openssl/crypto.cpp

```diff
@@ -103,6 +103,24 @@ auto engine_free_func = [](ENGINE *e) {
 	}
 };
 
+auto password_callback = [](char *buf, int size, int rwflag, void *u) {
+	// We'll only use this callback for reading passphrases, not for
+	// writing them.
+	assert(rwflag == 0);
+
+	if (u == nullptr) {
+		return 0;
+	}
+
+	// NB: buf is not expected to be null terminated.
+	char *const pass = static_cast<char *>(u);
+	strncpy(buf, pass, size);
+
+	const int len = static_cast<int>(strlen(pass));
```

Comment:
I see this is preexisting, but `pass` is not necessarily null terminated, so we need `strnlen` here. Probably not an issue, since the way `passphrase` is constructed using `c_str()` before being passed in, *is* null terminated, but just making sure that the comment matches what is actually being done.

Bonus: Then you don't need the conditional below anymore.